app/src/common/docs/BaseDocsBlock.tsx

```diff
@@ -1,136 +1,90 @@
-                const resProps = res.content.props;
-                const docPath = this.getPropsDocPath()[skin];
-                const docPathNorm = docPath.indexOf('.') === 0 ? docPath.substring(1) : docPath;
-                const props = resProps[docPathNorm];
-                /**
-                 * Keys in "public/docs/componentsPropsSet.json":
-                 * - always start from "/"
-                 * - are relative to the monorepo root.
-                 */
-                if (props) {
-                    this.propsDS = new ArrayDataSource({
-                        items: props,
-                        getId: (i) => i.name,
-                    });
-                    this.setState({ props: props });
-                }
-            });
-        }
+    private getSkin(): TSkin {
+        return getQuery('skin') || DEFAULT_SKIN;
     }
 
     abstract title: string;
     abstract renderContent(): React.ReactNode;
-    getPropsDocPath(): DocPath {
+    protected getPropsDocPath(): DocPath {
         return null;
     }
 
-    onTableStateChange = (newState: DataSourceState) => this.setState({ tableState: newState });
-    apiColumns: DataColumnProps<{ name: string; value: string; comment: string }>[] = [
-        {
-            key: 'name',
-            caption: 'NAME',
-            render: (prop) => <Text color="gray80">{prop.name}</Text>,
-            width: 200,
-            isSortable: true,
-        }, {
-            key: 'value',
-            caption: 'Type',
-            render: (prop) => (
-                <Text color="gray80">
-                    <span style={ { whiteSpace: 'pre-wrap' } }>{prop.value}</span>
-                </Text>
-            ),
-            width: 200,
-            isSortable: true,
-        }, {
-            key: 'comment',
-            caption: 'Description',
-            render: (prop) => <RichTextView htmlContent={ prop.comment } />,
-            width: 200,
-            grow: 1,
-        },
-    ];
+    protected getDocsGenType(): TDocsGenType | undefined {
+        return undefined;
+    }
 
     renderApiBlock() {
-        const view = this.propsDS.getView(this.state.tableState, this.onTableStateChange);
-
-        return (
-            <>
-                <RichTextView>
-                    <h2>Api</h2>
-                </RichTextView>
-                <DataTable
-                    value={ this.state.tableState }
-                    onValueChange={ this.onTableStateChange }
-                    columns={ this.apiColumns }
-                    getRows={ view.getVisibleRows }
-                    { ...view.getListProps() }
-                />
-            </>
-        );
+        const docsGenType = this.getDocsGenType();
+        let content = null;
+        if (docsGenType) {
+            content = (
+                <ApiReferenceItemTableForTypeRef showCode={ true } typeRef={ docsGenType } />
+            );
+        } else if (this.getPropsDocPath()) {
```

Comment:
I guess we already replace this.getPropsDocPath() in all places and cade remove this condition branch? This component is used only by us, so we shouldn't worry about other cases.

Reply:
done